apps/frontend/src/lib/components/Chat.svelte

```diff
@@ -0,0 +1,226 @@
+<script lang="ts">
+	import { Client } from '@langchain/langgraph-sdk';
+	import { streamAnswer } from '../utils/streamAnswer.js';
+	import ChatMessage from './ChatMessage.svelte';
+	import ToolMessage from './ToolMessage.svelte';
+	import ChatInput from './ChatInput.svelte';
+	import SuggestionCard from './SuggestionCard.svelte';
+
+	interface BaseMessage {
+		type: 'ai' | 'user' | 'tool';
+		text: string;
+	}
+
+	interface ToolMessageType extends BaseMessage {
+		type: 'tool';
+		tool_name: string;
+		payload?: any;
+		collapsed?: boolean;
+	}
+
+	type Message = BaseMessage | ToolMessageType;
+
+	interface Props {
+		client: Client;
+		assistantId: string;
+		threadId: string;
```

Comment:
I would not call this `client` as that abstracts away from the fact that it's a langgraph client. In (very near) future we might abstract even more, e.g. wrap this in some common parameter or even just make the entire function a parameter.